### DISCUSSION TITLE
Should rebase interactive (ri from status buffer on a recent commit) include the selected commit?

### DISCUSSION THREAD

**Discussion by star-szr** [NeogitOrg/neogit#862] [General]
I was watching a demo video of Magit and noticed that when running `ri` with the cursor on a recent commit, Magit includes that commit in the git-rebase-todo.

Put differently, it's the difference between:

- `git rebase -i hash` (current Neogit behaviour)
- `git rebase -i hash~` (Magit behaviour).

The Magit way seems more intuitive to me, what do you think? Maybe we should update this.

**Comment by star-szr** [comment]
I should also mention, this should likely apply when using the commit select view as well - for example if you hit `ri` from the top of the status buffer. The way you select the commit is not the important part, but that the result is what is expected.

**Comment by CKolkey** [comment]
Well spotted - agreed, we should go with the Magit behaviour.

**Comment by CKolkey** [comment]
Curious if there are other places like this that got overlooked. Maybe with reset and revert and such?

**Reply by star-szr** [reply]
Reset and revert seem fine, they reset to or revert the commit that is selected as I would expect. I think interactive rebase is a special case because for example `git rebase -i HEAD` will always give you noop in the `git-rebase-todo`, you would need to do at least `git rebase -i HEAD~` to get the current commit into `git-rebase-todo`.

**Reply by CKolkey** [reply]
Sounds good to me. Wanna do it?

**Reply by star-szr** [reply]
I’d be happy to try! Thanks.

**Reply by CKolkey** [reply]
Of course 😄 lemme know if you have any questions

**Reply by ten3roberts** [reply]
I am also very much for the `rebase interactive` includes the selected commit.

While following the behavior of git for commands named the same way is preferably, you need to keep in mind that git is not always the most intuitive, as is well known for many jokes. As such, small things like these which "do what you expect" are very appreciated

:heart:

**Reply by star-szr** [reply]
In case anyone else is following along, I'm posting here to say that I opened #883.

**Comment by star-szr** [comment]
Closing since this is in now! Thanks.
